src/mame/seta/speglsht.cpp

```diff
@@ -228,68 +226,69 @@ void speglsht_state::cop_w(offs_t offset, uint32_t data, uint32_t mem_mask)
 {
 	COMBINE_DATA(&m_cop_ram[offset]);
 
-	if(m_cop_ram[offset]&0x8000) //fix (sign)
+	if (m_cop_ram[offset] & 0x8000) // 16 bit signed to 32 bit
 	{
-		m_cop_ram[offset]|=0xffff0000;
+		m_cop_ram[offset] |= 0xffff0000;
 	}
 }
 
 //matrix * vector
 uint32_t speglsht_state::cop_r(offs_t offset)
 {
-	int32_t *cop=(int32_t*)&m_cop_ram[0];
+	int32_t *cop = (int32_t*)&m_cop_ram[0];
 
 	union
 	{
 		int32_t  a;
 		uint32_t b;
-	}temp;
+	} temp;
```

Comment:
This union is unnecessary.  Just use a local `int32_t` and cast to `uint32_t` when returning it.

```c++
	int32_t result = 0;
...
			result = ...;
...
	return uint32_t(result);
}
```